packages/libs/client/src/pact.ts

```diff
@@ -377,10 +377,12 @@ export class PactCommand
     return poll({ requestKeys: [this.requestKey] }, apiHost);
   }
 
-  public addSignatures(...sigs: { pubkey: string; sig: string }[]): this {
-    sigs.forEach(({ pubkey, sig }) => {
+  public addSignatures(
+    ...sigs: { pubKey: string; sig: string | null }[]
+  ): this {
+    sigs.forEach(({ pubKey: key, sig }) => {
       const foundSignerIndex = this.signers.findIndex(
-        ({ pubKey }) => pubKey === pubkey,
+        ({ pubKey }) => pubKey === key,
```

Comment:
you could also opt to not destructure the pubkey and compare like:

```
sigs.forEach(({ pubKey, sig }) => {
  const foundSignerIndex = this.signers.findIndex(
    (signer) => signer.pubKey === pubKey
```
That way you don't have to rename pubkey to a more abstract key

Reply:
normally I'm not for this but is this particular case it makes a lot of sense for the readability of the code